README.md

```diff
@@ -36,19 +37,80 @@ source .venv/bin/activate
 
 ## Usage
 
-After activation, you can import the ingestion stub:
+The following example demonstrates a full ingestion workflow using Spark, MinIO, and LinkML schema integration.
 
 ```python
-from data_lakehouse_ingest import ingest_from_config
-
-# Either a dict…
-cfg = {"input": {"uri": "s3a://demo/path"}, "target": {"table": "demo"}}
-print(ingest_from_config(cfg))  # True
+from data_lakehouse_ingest import data_lakehouse_ingest_config
+from data_lakehouse_ingest.logger import setup_logger
+from data_lakehouse_ingest.config_loader import ConfigLoader
+
+# Initialize Spark and MinIO -- works on Jupyter Hub
+spark = get_spark_session()
+minio_client = get_minio_client()
+
+# Set up structured logger
+pipeline_name = "pangenome"
+target_table = "genome"
+schema = "pangenome"
+
+logger = setup_logger(
+    pipeline_name=pipeline_name,
+    target_table=target_table,
+    schema=schema
+)
+
+# Option 1: Load config from MinIO
+cfg_path = "s3a://test-bucket/data-load/config-json/pangenome_ke_genome.json"
+
+# Option 2: Inline JSON config
+cfg_path = r'''
+{
+  "tenant": "pangenome_${dataset}",
+  "dataset": "pangenome_linkml",
+  "paths": {
+    "data_plane": "s3a://test-bucket/",
+    "bronze_base": "s3a://test-bucket/pangenome_ke/bronze",
+    "silver_base": "s3a://test-bucket/pangenome_linkml/silver"
+  },
+  "defaults": {
+    "tsv": { "header": true, "delimiter": "\t", "inferSchema": false }
+  },
+  "tables": [
+    {
+      "name": "genome",
+      "linkml_schema": "s3a://test-bucket/data-load/linkml-schema/genome_schema.yml",
+      "schema_sql": null,
+      "partition_by": null,
+      "bronze_path": "s3a://test-bucket/pangenome_ke/bronze/table_genome_V1.1.tsv"
+    }
+  ]
+}
+'''
+
+# Run ingestion
+report = data_lakehouse_ingest_config(
+    cfg_path,
+    spark=spark,
+    logger=logger,
+    minio_client=minio_client
+)
+
+print(report)
 
-# …or a JSON file path
-print(ingest_from_config("/path/to/config.json"))  # True
 ```
 
+---
+
+## 🧠 How it works
```

Comment:
The emojis make it a bit obvious that this was created by genAI -- YMMV but I find them a bit tacky.